commit 9049ccd46f1f39166c99861c2c59d01ce3e20fd6
Author: Lars Ellenberg <lars.ellenberg@linbit.com>
Date:   Thu Dec 20 17:23:40 2018 +0100

    drbd: don't retry connection if peers do not agree on "authentication" settings
    
    emma: "Unexpected data packet AuthChallenge (0x0010)"
     ava: "expected AuthChallenge packet, received: ReportProtocol (0x000b)"
          "Authentication of peer failed, trying again."
    
    Pattern repeats.
    
    There is no point in retrying the handshake,
    if we expect to receive an AuthChallenge,
    but the peer is not even configured to expect or use a shared secret.
    
    Signed-off-by: Lars Ellenberg <lars.ellenberg@linbit.com>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/drivers/block/drbd/drbd_receiver.c b/drivers/block/drbd/drbd_receiver.c
index 3a0fe357b68b..02a327891568 100644
--- a/drivers/block/drbd/drbd_receiver.c
+++ b/drivers/block/drbd/drbd_receiver.c
@@ -5332,7 +5332,7 @@ static int drbd_do_auth(struct drbd_connection *connection)
 	if (pi.cmd != P_AUTH_CHALLENGE) {
 		drbd_err(connection, "expected AuthChallenge packet, received: %s (0x%04x)\n",
 			 cmdname(pi.cmd), pi.cmd);
-		rv = 0;
+		rv = -1;
 		goto fail;
 	}